src/middlewared/middlewared/api/base/server/app.py

```diff
@@ -0,0 +1,18 @@
+import logging
+import uuid
+
+from middlewared.auth import SessionManagerCredentials
+from middlewared.utils.origin import Origin
+
+logger = logging.getLogger(__name__)
+
+
+class App:
+    def __init__(self, origin: Origin):
+        self.origin = origin
+        self.session_id = str(uuid.uuid4())
+        self.authenticated = False
+        self.authenticated_credentials: SessionManagerCredentials | None = None
+        self.py_exceptions = False
+        self.websocket = False
+        self.rest = False
```

Comment:
If `websocket` and `rest` are mutually exclusive, only one is necessary. It might be even better to cut both and just use `isinstance(app, RpcWebSocketApp)` and `isinstance(app, Application)`.

Reply:
Or we could have an enum for session type

Reply:
It's like this for backwards compatibility with the existing code. This will be refactored when existing server code goes away.